src/components/PrimaryPanes/SourcesTree.js

```diff
@@ -144,6 +145,7 @@ class SourcesTree extends Component<Props, State> {
     // only when a new source is added
     if (nextProps.sources != this.props.sources) {
       this.setState(
+        /*
         updateTree({
```

Comment:
Ideally we can use this function but it's causing problems atm.

Reply:
By doing more testing, I'm seeing that updateTree is a necessity, because directories from previous pages sometimes display when they shouldn't.